# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Barrier.cpp
# Commit: 28829816a629
# Full Hash: 28829816a629b28125db3b4c65c5e22518e167e8
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-09 21:37:17
# Regressor Bug: 1643586
# File Overlap Count: 1
# Description:
#   Bug 1643586 - Part 6: Stop using JSContext for GC helper thread tasks r=jandem
#   
#   This patch moves the debug-only GC state from JSContext to JSFreeOp. Support
#   functions for this move to FreeOp.h and most RAII classes move to GCInternals.h.
#   
# ==============================================================================

diff -r 410d1b57bb09 -r 28829816a629 js/src/gc/Barrier.cpp
--- a/js/src/gc/Barrier.cpp	Wed Mar 09 10:17:07 2022 +0000
+++ b/js/src/gc/Barrier.cpp	Wed Mar 09 10:17:08 2022 +0000
@@ -67,36 +67,6 @@
   return jcx && jcx->inIonBackend();
 }
 
-bool CurrentThreadIsGCMarking() {
-  JSContext* cx = MaybeGetJSContext();
-  return cx && cx->gcUse == JSContext::GCUse::Marking;
-}
-
-bool CurrentThreadIsGCSweeping() {
-  JSContext* cx = MaybeGetJSContext();
-  return cx && cx->gcUse == JSContext::GCUse::Sweeping;
-}
-
-bool CurrentThreadIsGCFinalizing() {
-  JSContext* cx = MaybeGetJSContext();
-  return cx && cx->gcUse == JSContext::GCUse::Finalizing;
-}
-
-bool CurrentThreadIsTouchingGrayThings() {
-  JSContext* cx = MaybeGetJSContext();
-  return cx && cx->isTouchingGrayThings;
-}
-
-AutoTouchingGrayThings::AutoTouchingGrayThings() {
-  TlsContext.get()->isTouchingGrayThings++;
-}
-
-AutoTouchingGrayThings::~AutoTouchingGrayThings() {
-  JSContext* cx = TlsContext.get();
-  MOZ_ASSERT(cx->isTouchingGrayThings);
-  cx->isTouchingGrayThings--;
-}
-
 #endif  // DEBUG
 
 // Tagged pointer barriers